core/app/paginators/spree/products/paginate.rb

```diff
@@ -0,0 +1,19 @@
+module Spree
```

Comment:
Let's move it under `Spree::Shared` namespace and use it for both products and taxons.
Also - what do you think about passing collection in hash, e.g `params[:collection]` ?

Reply:
For me looks better when collection go as separate parameter, more naturally.

Reply:
Okay, sure